addons/mail/static/tests/chatter_tests.js

```diff
@@ -312,6 +312,172 @@ QUnit.test('list activity widget: open dropdown', async function (assert) {
+    assert.expect(11);
+
+    const currentUser = this.data['res.users'].records.find(user =>
+        user.id === this.data.currentUserId
+    );
+    Object.assign(currentUser, {
+        activity_ids: [1, 4],
+        activity_state: 'today',
+        activity_summary: 'Call with Al',
+        activity_type_id: 3,
+    });
+    this.data['mail.activity'].records.push(
+        {
+            id: 1,
+            display_name: "Call with Al",
+            date_deadline: moment().format("YYYY-MM-DD"), // now
+            can_write: true,
+            state: "today",
+            user_id: this.data.currentUserId,
+            create_uid: this.data.currentUserId,
+            activity_type_id: 3,
+        },
+        {
+            id: 4,
+            display_name: "Meet FP",
+            date_deadline: moment().add(1, 'day').format("YYYY-MM-DD"), // tomorrow
+            can_write: true,
+            state: "planned",
+            user_id: this.data.currentUserId,
+            create_uid: this.data.currentUserId,
+            activity_type_id: 1,
+        }
+    );
+
+    const { env, widget: list } = await start({
+        hasView: true,
+        View: ListView,
+        model: 'res.users',
+        data: this.data,
+        arch: `
+            <list>
+                <field name="foo"/>
+                <field name="activity_ids" widget="list_activity"/>
+            </list>`,
+        mockRPC: function (route, args) {
+            assert.step(args.method || route);
+            if (args.method === 'action_feedback_schedule_next'){
+                assert.strictEqual(args.args.length, 1);
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 1);
+                return Promise.resolve({serverGeneratedAction: true});
+            }
+            return this._super(route, args);
+        },
+        intercepts: {
+            do_action: function (ev) {
+                var action = ev.data.action;
+                if (action.serverGeneratedAction) {
+                    assert.step('serverGeneratedAction');
+                }
+            },
+            switch_view: () => assert.step('switch_view'),
+        },
+    });
+
+    await testUtils.dom.click(list.$('.o_data_cell:first'));
+
+    assert.step('open dropdown');
+    await testUtils.dom.click(list.$('.o_activity_btn span')); // open the popover
+    await testUtils.dom.click(list.$('.o_mark_as_done:first')); // mark the first activity as done
+    await testUtils.fields.triggerKeydown(list.$('.o_activity_popover_done_next'), 'enter'); //confirm
+
+    assert.verifySteps([
+        '/web/dataset/search_read',
+        '/mail/init_messaging',
+        'switch_view',
+        'open dropdown',
+        'activity_format',
+        'action_feedback_schedule_next',
+        'serverGeneratedAction'
+    ]);
+
+    list.destroy();
+});
```

Comment:
All the same remarks as first test, try to keep it as short as possible and to the point (only asserting that the keyboard shortcut works and nothing else).